packages/esm-patient-common-lib/src/dashboards/MenuIcons.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import {
+  ActivityIcon,
+  ShoppingCartIcon,
+  MedicationIcon,
+  ChartAverageIcon,
+  CalendarHeatMapIcon,
+  WarningIcon,
+  ListCheckedIcon,
+  DocumentAttachmentIcon,
+  EventScheduleIcon,
+  ReportIcon,
+  SyringeIcon,
+  ProgramsIcon,
+} from '@openmrs/esm-framework';
+import styles from './dashboard-extension.scss';
+
+export type MenuTitle = keyof typeof MenuIcons;
+
+export const MenuIcons = {
+  'Patient Summary': ReportIcon,
+  'Vitals & Biometrics': ActivityIcon,
+  Medications: MedicationIcon,
+  Orders: ShoppingCartIcon,
+  Results: ChartAverageIcon,
+  Visits: CalendarHeatMapIcon,
+  Allergies: WarningIcon,
+  Conditions: ListCheckedIcon,
+  Immunizations: SyringeIcon,
+  Attachments: DocumentAttachmentIcon,
+  Programs: ProgramsIcon,
+  Appointments: EventScheduleIcon,
+} as const;
+
+interface IconProps {
+  title: MenuTitle;
+}
+
+export const IconRenderer = ({ title }: IconProps) => {
```

Comment:
So... title is an arbitrary string and you've used `as any` to work around Typescript's complaining about the mismatch. Just type things more weakly here, i.e., `title` should just be a string.

Reply:
Design-wise, I still don't love the hard-coding of menu icons here, since it makes this completely unusable except for a few dashboards we have foreknowledge of. Since this PR started, I added a more generic `MaybeIcon` component that takes a string like `omrs-report-icon` and renders the icon if that exists. Instead of relying on the `title` to chose which `icon`, I suggest we make the icon part of the props that get passed to the `DashboardExtension` via the extension meta. This allows us to support more than just a small list of known panels.